File: Java/766.java
__________________________________________________________________________________________________
sample 1 ms submission
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {
        for (int i = 0; i < matrix[0].length; i++) {
            if (!check(matrix, matrix[0][i], 0, i)) {
                return false;
            }
        }
        for (int i = 0; i < matrix.length; i++) {
            if (!check(matrix, matrix[i][0], i, 0)) {
                return false;
            }
        }
        return true;
      }

    private boolean check(int[][] matrix, int expectValue, int row, int col) {
        if (row >= matrix.length || col >= matrix[0].length) {
            return true;
        }
        if (matrix[row][col] != expectValue) {
            return false;
        }
        return check(matrix, expectValue, row + 1, col + 1);
    }
}
__________________________________________________________________________________________________
sample 40364 kb submission
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {
        for(int i = 1; i< matrix.length; i++)
        {
            for(int j= 1; j< matrix[0].length; j++)
            {
                if(matrix[i][j] != matrix[i-1][j-1])
                    return false;
            }
        }
        return true;
    }
}
__________________________________________________________________________________________________

File: Python3/766.py
__________________________________________________________________________________________________
sample 84 ms submission
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        
        for i in range(len(matrix)-1):
            if (matrix[i][0:len(matrix[0])-1] != 
                matrix[i+1][1:len(matrix[0])]):
                return False
        
        return True
__________________________________________________________________________________________________
sample 13028 kb submission
import sys

class Solution:
    def isToeplitzMatrix(self, A: List[List[int]]) -> bool:
        m, n = len(A), len(A[0])
        for i in range(m):
            for j in range(n):
                if A[i][j] == -sys.maxsize:
                    continue
                k, l = i+1, j+1
                while k < m and l < n:
                    if A[k-1][l-1] == A[k][l]:
                        A[k-1][l-1] = -sys.maxsize
                        k, l = k+1, l+1                        
                    else:
                        return False
        return True
__________________________________________________________________________________________________
